### PR TITLE
!!!TASK: Rename 'uploadExtensionBlacklist' setting to 'extensionsBlockedFromUpload'

### PR DESCRIPTION
**Pull request by albe**
As part of our move to getting rid of bad wording in our code base, this changes the setting `Neos.Flow.resource.uploadExtensionsBlacklist` to the much more descriptive `Neos.Flow.resource.extensionsBlockedFromUpload`.

This is breaking as it changes the path to a setting, but is covered by a code migration.

Related to #2024

### CODE REVIEW

**Comment by albe** [comment]
I do like `extensionsBlockedFromUpload`

File: Neos.Flow/Classes/ResourceManagement/ResourceManager.php [hunk 1]
```diff
@@ -625,7 +625,7 @@ protected function prepareUploadedFileForImport(array $uploadInfo)
             throw new Exception('The given upload file "' . strip_tags($pathInfo['basename']) . '" was not uploaded through PHP. As it could pose a security risk it cannot be imported.', 1422461503);
         }
 
-        if (isset($pathInfo['extension']) && array_key_exists(strtolower($pathInfo['extension']), $this->settings['uploadExtensionBlacklist']) && $this->settings['uploadExtensionBlacklist'][strtolower($pathInfo['extension'])] === true) {
+        if (isset($pathInfo['extension']) && array_key_exists(strtolower($pathInfo['extension']), $this->settings['uploadExtensionsExcluded']) && $this->settings['uploadExtensionsExcluded'][strtolower($pathInfo['extension'])] === true) {
```
Comment (albe) [inline Neos.Flow/Classes/ResourceManagement/ResourceManager.php:628]:
```suggestion
        if (isset($pathInfo['extension']) && array_key_exists(strtolower($pathInfo['extension']), $this->settings['extensionsBlockedFromUpload']) && $this->settings['extensionsBlockedFromUpload'][strtolower($pathInfo['extension'])] === true) {
```

File: Neos.Flow/Configuration/Settings.Resource.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ Neos:
     resource:
       # A list of filename extensions that must not be uploaded through the resource
       # management.
-      uploadExtensionBlacklist: &uploadExtensionBlacklist
+      uploadExtensionsExcluded: &uploadExtensionsExcluded
```
Comment (albe) [inline Neos.Flow/Configuration/Settings.Resource.yaml:10]:
```suggestion
      extensionsBlockedFromUpload: &extensionsBlockedFromUpload
```

File: Neos.Flow/Configuration/Settings.Resource.yaml [hunk 3]
```diff
@@ -62,7 +62,7 @@ Neos:
           targetOptions:
             path: '%FLOW_PATH_WEB%_Resources/Static/Packages/'
             baseUri: '_Resources/Static/Packages/'
-            extensionBlacklist: *uploadExtensionBlacklist
+            excludedExtensions: *uploadExtensionsExcluded
```
Comment (albe) [inline Neos.Flow/Configuration/Settings.Resource.yaml:65]:
```suggestion
            excludedExtensions: *extensionsBlockedFromUpload
```

File: Neos.Flow/Configuration/Settings.Resource.yaml [hunk 4]
```diff
@@ -73,7 +73,7 @@ Neos:
           targetOptions:
             path: '%FLOW_PATH_WEB%_Resources/Persistent/'
             baseUri: '_Resources/Persistent/'
-            extensionBlacklist: *uploadExtensionBlacklist
+            excludedExtensions: *uploadExtensionsExcluded
```
Comment (albe) [inline Neos.Flow/Configuration/Settings.Resource.yaml:76]:
```suggestion
            excludedExtensions: *extensionsBlockedFromUpload
```

File: Neos.Flow/Resources/Private/Schema/Settings/Neos.Flow.resource.schema.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 type: dictionary
 additionalProperties: false
 properties:
-  'uploadExtensionBlacklist':
+  'uploadExtensionsExcluded':
```
Comment (albe) [inline Neos.Flow/Resources/Private/Schema/Settings/Neos.Flow.resource.schema.yaml:4]:
```suggestion
  'extensionsBlockedFromUpload':
```

**Comment by albe** [comment]
What's with those style-ci errors lately? Did the CGL change?

Anyway, this is ready now IMO

**Comment by kdambekalns** [comment]
> What's with those style-ci errors lately? Did the CGL change?

I don't know. My hope right now is, that once we have the code clean, it'll find only "real, new" errors…

**Comment by albe** [comment]
friendly ping @bwaidelich :) I hope your valid concerns have been met sufficiently

**Review by bwaidelich — APPROVED** [review]
Awesome, thank you!

**Comment by bwaidelich** [comment]
@albe you have added this PR to the 6.0/7.0 project board, but we decided only to work with issues there (in order to keep it easier to maintain and to have a nice changelog that links to the higher level issue descriptions). Is there a corresponding ticket already?

**Comment by albe** [comment]
Not really sure what should be in that issue in this case tbh. :/ "We need to fix the wording of things" with the two tasks of #2024 and this?

**Comment by bwaidelich** [comment]
> Not really sure what should be in that issue in this case

IMO it's less about the wording in the issue but more about coordination and overview. I created #2124 and linked this PR and the other one with it.

@albe are you aware of more related exclusive wording or can we close the issue for now?

**Comment by albe** [comment]
The only thing I have on my mind still right now (apart from the Neos counterparts) is the default branch name "master", but I think that needs some decission finding in the team first.

**Comment by bwaidelich** [comment]
> The only thing I have on my mind still right now (apart from the Neos counterparts) is the default branch name "master", but I think that needs some decission finding in the team first.

Aah, right! Thanks for the reminder and for updating the issue. I'd say we keep it open for now and see if we can tackle it in time for the next release!

**Pull request merged**